app/main.py

```diff
@@ -1,54 +1,36 @@
-import json
-import xml.etree.ElementTree as ET
-
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.book_display import ConsoleDisplayBook, ReverseDisplayBook
+from app.book_print import ConsolePrintBook, ReversePrintBook
+from app.book_serializer import JsonSerializer, XmlSerializer
+
+data = {
+    "display": {
+        "console": ConsoleDisplayBook, "reverse": ReverseDisplayBook,
+    },
+    "print": {
+        "console": ConsolePrintBook, "reverse": ReversePrintBook,
+    },
+    "serialize": {
+        "json": JsonSerializer, "xml": XmlSerializer,
+    }
+}
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            data.get("display").get(method_type)(book).display()
         elif cmd == "print":
-            book.print_book(method_type)
+            data.get("print").get(method_type)(book).print_book()
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            return data.get("serialize").get(method_type)(book).serialize()
```

Comment:
you removed initial errors of unknown types. let's handle unknown cmd and method_type